tests/Unit/Helpers/Evolution/DiscontinuousGalerkin/BoundaryCorrections.hpp

```diff
@@ -0,0 +1,317 @@
+      call_dg_package_data(make_not_null(&interior_package_data_opposite_signs),
+                           correction, interior_fields_on_face, volume_data,
+                           exterior_unit_normal_covector, mesh_velocity);
+      Variables<dt_variables_tags> zero_boundary_correction{
+          face_mesh.number_of_grid_points()};
+      call_dg_boundary_terms(make_not_null(&zero_boundary_correction),
+                             correction, interior_package_data,
+                             interior_package_data_opposite_signs,
+                             ::dg::Formulation::StrongInertial);
+      Variables<dt_variables_tags> expected_zero_boundary_correction{
+          face_mesh.number_of_grid_points(), 0.0};
+      tmpl::for_each<dt_variables_tags>([&expected_zero_boundary_correction,
+                                         &zero_boundary_correction](
+                                            auto dt_variables_tag_v) noexcept {
+        using dt_variables_tag = tmpl::type_from<decltype(dt_variables_tag_v)>;
+        const std::string tag_name = db::tag_name<dt_variables_tag>();
+        CAPTURE(tag_name);
+        Approx custom_approx = Approx::custom().epsilon(1.e-12).scale(1.0);
+        CHECK_ITERABLE_CUSTOM_APPROX(
+            get<dt_variables_tag>(zero_boundary_correction),
+            get<dt_variables_tag>(expected_zero_boundary_correction),
+            custom_approx);
+      });
+    }
+  } else if (dg_formulation == ::dg::Formulation::WeakInertial) {
+    INFO(
+        "Checking that swapping the two sides results in an overall minus "
+        "sign.");
+    Variables<dt_variables_tags> reverse_side_boundary_corrections{
+        face_mesh.number_of_grid_points()};
+    call_dg_boundary_terms(make_not_null(&reverse_side_boundary_corrections),
+                           correction, exterior_package_data,
+                           interior_package_data, dg_formulation);
+    // Reversing the sign and having them be equal is a statement of "what goes
+    // out one element enters the other".
+    reverse_side_boundary_corrections *= -1.0;
+    tmpl::for_each<flux_variables>([&boundary_corrections,
+                                    &reverse_side_boundary_corrections](
+                                       auto flux_variable_tag_v) {
+      using flux_variable_tag = tmpl::type_from<decltype(flux_variable_tag_v)>;
+      const std::string tag_name = db::tag_name<flux_variable_tag>();
+      CAPTURE(tag_name);
+      CHECK_ITERABLE_APPROX(
+          get<::Tags::dt<flux_variable_tag>>(reverse_side_boundary_corrections),
+          get<::Tags::dt<flux_variable_tag>>(boundary_corrections));
+    });
+  } else {
+    ERROR("DG formulation must be StrongInertial or WeakInertial, not "
+          << dg_formulation);
+  }
+}
+}  // namespace detail
+
+/*!
+ * \ingroup TestingFrameworkGroup
+ * \brief Checks that the boundary correction is conservative and that for
+ * smooth solutions the correction is zero.
```

Comment:
[clarity nitpick] "the correction is zero" -> "the strong-form correction is zero" ?